File: internal/domain/accounts.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

const (
	AccountStatusActive   = "active"
	AccountStatusArchived = "archived"
)

type Account struct {
	ID        string    `json:"id" db:"id"`
	FirstName string    `json:"first_name" db:"first_name"`
	LastName  string    `json:"last_name" db:"last_name"`
	Age       int       `json:"age" db:"age"`
	Email     string    `json:"email" db:"email"`
	Avatar     string    `json:"avatar" db:"avatar"`
	Status     string    `json:"status" db:"status"`
	DoctorID   string    `json:"doctor_id" db:"-"`
	HospitalID string    `json:"hospital_id" db:"-"`
	CreatedAt  time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
}

func NewAccount(firstName string, lastName string, age int, email string, avatar string) *Account {
	now := time.Now().UTC()
	return &Account{
		ID:        uuid.NewString(),
		FirstName: firstName,
		LastName:  lastName,
		Age:       age,
		Email:     email,
		Avatar:    avatar,
		Status:    AccountStatusActive,
		CreatedAt: now,
		UpdatedAt: now,
	}
}

File: _frontend/src/domain/accounts.ts
export enum AccountStatus {
  Active = "active",
  Archived = "archived",
}

export interface Account {
  id: string;
  first_name: string;
  last_name: string;
  age: number;
  email: string;
  avatar: string;
  status: AccountStatus;
  doctor_id?: string;
  hospital_id?: string;
  created_at: string;
  updated_at: string;
}

// getAccountDisplayName -
// getAccountDisplayName - Returns the full name of an account.
export const getAccountDisplayName = (account: Account): string =>
  `${account.first_name} ${account.last_name}`;
